src/appleseed-max-impl/appleseedinteractive/interactivetilecallback.cpp

```diff
@@ -0,0 +1,53 @@
+
+// Interface header.
+#include "appleseedinteractive/interactivetilecallback.h"
+
+namespace asr = renderer;
+
+#define WM_TRIGGER_CALLBACK WM_USER + 4764
+
+InteractiveTileCallback::InteractiveTileCallback(
+    Bitmap*                         bitmap,
+    IIRenderMgr*                    iimanager,
+    asr::IRendererController*       render_controller
+    )
+    : TileCallback(bitmap, 0)
+    , m_bitmap(bitmap)
+    , m_iimanager(iimanager)
+    , m_renderer_ctrl(render_controller)
+{
+}
+
+void InteractiveTileCallback::update_window()
```

Comment:
this function is called from ui thread. I block the renderer thread until this function completes in order to guarantee m_iimanager still exists. m_ui_promise is class member, I'm not sure if it's ok. It is assigned new promise object on each frame_end.

Reply:
Get rid of this method, inline in `update_caller()`.